webcompat/helpers.py

```diff
@@ -368,3 +374,47 @@ def extract_url(issue_body):
     else:
         url = ""
     return url
+
+
+def proxy_request(method, path, params=None, headers=None, data=None):
+    '''Make a GitHub API request with a bot's OAuth token.
+
+    Necessary for non-logged in users.
+    * `path` will be appended to the end of the API_URI.
+    * Optionally pass in POST data via the `data` arg.
+    * Optionally point to a different URI via the `uri` arg.
+    * Optionally pass in HTTP headers to forward.
+    '''
+    # Merge passed in headers with AUTH_HEADERS, and add the etag of the
+    # request, if it exists, to be sent back to GitHub.
+    auth_headers = AUTH_HEADERS.copy()
+    if headers:
+        auth_headers.update(headers)
+    # Grab the correct Requests request method
+    req = getattr(requests, method)
+    # It's expected that path *won't* start with a leading /
+    # https://api.github.com/repos/{0}
+    resource_uri = API_URI + path
+    return req(resource_uri, data=data, params=params, headers=auth_headers)
+
+
+def api_request(method, path, params=None, data=None):
+    '''Helper to abstract talking to the  GitHub API.
+
+    This handles both logged-in and proxied requests.
+    This returns a tuple, which if returned from a view function will
+    be automatically turned into a response. But if you return it elsewhere,
+    pay attention.
+    '''
+    request_headers = get_request_headers(g.request_headers)
+    if g.user:
+        request_method = github.raw_request
+    else:
+        request_method = proxy_request
+    resource = request_method(method, path, headers=request_headers,
+                              params=params, data=data)
+    if resource.status_code != 404:
+        return (resource.content, resource.status_code,
+                get_response_headers(resource))
```

Comment:
So I was saying earlier returning a tuple maybe a bit difficult in some circumstances in the code. I suggested a dict or a namedtuple. (This can come later, small improvements.)